src/main/java/payapal/Perfection.java

```diff
@@ -0,0 +1,42 @@
+package main.java.payapal;
+
+public class Perfection {
+
+    private static Perfection perf;
+
+    public synchronized static Perfection getPerf() {
```

Comment:
I think it doesn't make much sense to have this class as a singleton since its sole public method is static and the class does not hold or manipulate any kind of state.

Having said that, I would suggest a few improvements to the singleton implementation which could come in handy for other candidate classes:

* A private constructor would be needed so that nobody can instantiate the class from the outside
* A Reentrant Lock could be used instead of syncronized keyword since it's more performant.
* A single sincronization or locking is imperfect, a double checked locking mechanism should be implemented so that after the first null check has passed the lock is checked again, otherwise there could be cases where the class is instantiated more than once.